File: OP_laba5_part2_c++/src/ParentString.cpp
#include "ParentString.h"

int ParentString::GetLength()
{
    return sizeof(this->value);
}

char *ParentString::GetValue()
{
    return this->value;
}

ParentString::ParentString(char *value)
{
    this->value = value;
}

void ParentString::Sort(){}

File: OP_laba5_part2_sharp/OP_laba5_part2_sharp/ParentString.cs
using System;

namespace OP_laba5_part2_sharp
{
    public class ParentString
    {
        protected char[] StrValue { get; set; }

        public ParentString(params char[] str)
        {
            StrValue = str;
        }
        
        //Методи отримання довжини та значень не віртуальні бо не має сенсу
        public int GetLength() => StrValue.Length;
        public char[] GetValue() => StrValue;

        public virtual void Sort() {}
        
    }
}